spec/lib/video_info/providers/youtube_api_spec.rb

```diff
@@ -0,0 +1,392 @@
+    end
+  end
+
+  describe "#available?" do
+    context "with valid video", :vcr do
+      subject { VideoInfo.new('http://www.youtube.com/watch?v=mZqGqE0D0n4') }
+
+      describe '#available?' do
+        subject { super().available? }
+        it { is_expected.to be_truthy }
+      end
+    end
+
+    context "with 'video is unavailable' video", :vcr do
+      subject { VideoInfo.new('http://www.youtube.com/watch?v=SUkXvWn1m7Q') }
+
+      describe '#available?' do
+        subject { super().available? }
+        it { is_expected.to be_falsey }
+      end
+
+      describe '#provider' do
+        subject { super().provider }
+        it { is_expected.to eq 'YouTube' }
+      end
+
+      describe '#video_id' do
+        subject { super().video_id }
+        it { is_expected.to eq 'SUkXvWn1m7Q' }
+      end
+
+      describe '#url' do
+        subject { super().url }
+        it { is_expected.to eq 'http://www.youtube.com/watch?v=SUkXvWn1m7Q' }
+      end
+
+      describe '#embed_url' do
+        subject { super().embed_url }
+        it { is_expected.to eq '//www.youtube.com/embed/SUkXvWn1m7Q' }
+      end
+
+      describe '#embed_code' do
+        subject { super().embed_code }
+        it { is_expected.to eq '<iframe src="//www.youtube.com/embed/SUkXvWn1m7Q" frameborder="0" allowfullscreen="allowfullscreen"></iframe>' }
+      end
+
+      describe '#title' do
+        subject { super().title }
+        it { is_expected.to eq nil }
+      end
+
+      describe '#description' do
+        subject { super().description }
+        it { is_expected.to eq nil }
+      end
+
+      describe '#keywords' do
+        subject { super().keywords }
+        it { is_expected.to be_nil }
+      end
+
+      describe '#duration' do
+        subject { super().duration }
+        it { is_expected.to eq 0 }
+      end
+
+      describe '#date' do
+        subject { super().date }
+        it { is_expected.to eq nil }
+      end
+
+      describe '#thumbnail_small' do
+        subject { super().thumbnail_small }
+        it { is_expected.to eq 'https://i.ytimg.com/vi/SUkXvWn1m7Q/default.jpg' }
+      end
+
+      describe '#thumbnail_medium' do
+        subject { super().thumbnail_medium }
+        it { is_expected.to eq 'https://i.ytimg.com/vi/SUkXvWn1m7Q/mqdefault.jpg' }
+      end
+
+      describe '#thumbnail_large' do
+        subject { super().thumbnail_large }
+        it { is_expected.to eq 'https://i.ytimg.com/vi/SUkXvWn1m7Q/hqdefault.jpg' }
+      end
+
+      describe '#view_count' do
+        subject { super().view_count }
+        it { is_expected.to be == 0 }
+      end
+    end
+
+    context "with 'video no longer available due to a copyright claim' video", :vcr do
+      subject { VideoInfo.new('http://www.youtube.com/watch?v=ffClNhwx0KU') }
+
+      describe '#available?' do
+        subject { super().available? }
+        it { is_expected.to be_falsey }
+      end
+    end
+  end
+
+  context "with video mZqGqE0D0n4", :vcr do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.